src/system/dlt-system-filetransfer.c

```diff
@@ -274,9 +274,9 @@ int send_one(char *src, FiletransferOptions const *opts, int which)
         return -1;
     }
 
-    char *fdir = strndup(src, PATH_MAX);
-    MALLOC_ASSERT(fdir);
-    fdir = dirname(fdir);/*dirname overwrites its argument anyway */
+    char *src_copy = strndup(src, PATH_MAX);
+    MALLOC_ASSERT(src_copy);
+    char* fdir = dirname(src_copy);/*dirname overwrites its argument anyway, depending on argument the returned address might change */
```

Comment:
Please keep the line in 80 columns.

Reply:
Yes it is only a memory leak, which can happen.

Reply:
> 
> 
> Please keep the line in 80 columns.
Mh, line 315 (not touched by me) is even longer. Shall I reformat them too?

Reply:
> > Please keep the line in 80 columns.
> > Mh, line 315 (not touched by me) is even longer. Shall I reformat them too?

Hum, that is also true. But reformatting them is too much and more than what the commit itself you fix, so please modify only your lines. I can do those kind of reformatting for all the files. Thanks for reminding me of this.

Reply:
Line changed